uniffi_macros/src/error.rs

```diff
@@ -157,7 +157,31 @@ fn flat_error_ffi_converter_impl(
             }
 
         }
-    });
+    } else {
+        quote! {
+            // Lifting flat errors is not currently supported, but we still define the trait so
+            // that dicts containing flat errors don't cause compile errors (see ReturnOnlyDict in
+            // coverall.rs).
+            //
+            // Note: it would be better to not derive `Lift` for dictionaries containing flat
+            // errors, but getting the trait bounds and derived impls there would be much harder.
+            // For now, we just fail at runtime.
+            #[automatically_derived]
+            unsafe #lift_impl_spec {
+                type FfiType = ::uniffi::RustBuffer;
+
+                fn try_read(buf: &mut &[::std::primitive::u8]) -> ::uniffi::deps::anyhow::Result<Self> {
+                    panic!("Can't lift flat errors")
```

Comment:
is it worth opening an issue around getting these errors at buildtime instead of a runtime panic?

Reply:
Good point, I made https://github.com/mozilla/uniffi-rs/issues/1850 for this.